### PR TITLE
Add step argument to allow equally-spaced overlapping windows.

### LINKED ISSUE #790: Can't make equally spaced overlapping windows
I can't seem to do this with `window_by_position`. 

I looks like the  `window_start_position` argument implicitly expects the specified positions match the contigs of the `variant_contig` array. The contig bounds are calculated [here](https://github.com/pystatgen/sgkit/blob/main/sgkit/window.py#L229) from the `variant_contig` array. The bounds of a given contig (start and stop) are the indices corresponding to the first and last variants on the contig. These bounds are then used [here](https://github.com/pystatgen/sgkit/blob/main/sgkit/window.py#L291) to slice the provided `window_start_positions` and therefore assume that `window_start_positions` match the contigs of `variant_contig`. This assumption is true if  you use `window_start_positions=variant_position` but unlikely to be true for a custom array.

There doesn't seem to be a way to make overlapping windows without using the  `window_start_position` argument.

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #790

### CODE REVIEW

File: sgkit/window.py [hunk 1]
```diff
@@ -113,6 +114,9 @@ def window_by_position(
         Genotype call dataset.
     size
         The window size, measured by number of base pairs.
+    step
```
[the remark this answers is not in the document]
Comment (tomwhite) [reply sgkit/window.py:117]: I've changed it to raise a ValueError. Also added a test and some documentation.

[the remark this answers is not in the document]
Comment (jeromekelleher) [reply sgkit/window.py:117]: Probably safer all right.

**Comment by jeromekelleher** [comment]
Happy to merge - @timothymillar do you want to give your formal approval here if you've looked at the PR?

**Review by timothymillar — APPROVED** [review]
LGTM, thanks @tomwhite

**Pull request merged**